packages/media-utils/src/utils/types.ts

```diff
@@ -0,0 +1,207 @@
+/**
+ * A media attachment object in a REST API context.
+ *
+ * Simplified version of what's defined in the wp-types package.
+ *
+ * @see https://www.npmjs.com/package/wp-types
+ */
+interface WP_REST_API_Attachment {
+	/**
+	 * Unique identifier for the attachment.
+	 */
+	id: number;
+	/**
+	 * The ID of the featured media for the post.
+	 */
+	featured_media: number;
+	/**
+	 * URL to the attachment.
+	 */
+	link: string;
+	/**
+	 * The date the attachment was published, in the site's timezone.
+	 */
+	date: string;
+	/**
+	 * The date the attachment was published, as GMT.
+	 */
+	date_gmt: string;
+	/**
+	 * The date the attachment was last modified, in the site's timezone.
+	 */
+	modified: string;
+	/**
+	 * The date the attachment was last modified, as GMT.
+	 */
+	modified_gmt: string;
+	/**
+	 * An alphanumeric identifier for the attachment unique to its type.
+	 */
+	slug: string;
+	/**
+	 * A named status for the attachment.
+	 */
+	status: string;
+	/**
+	 * Type of Post for the attachment.
+	 */
+	type: 'attachment';
+	/**
+	 * Alternative text to display when attachment is not displayed.
+	 */
+	alt_text: string;
+	/**
+	 * The attachment caption.
+	 */
+	caption: {
+		/**
+		 * Caption for the attachment, as it exists in the database. Only present when using the 'edit' context.
+		 */
+		raw?: string;
+		/**
+		 * HTML caption for the attachment, transformed for display.
+		 */
+		rendered: string;
+	};
+	/**
+	 * The attachment description.
+	 */
+	description: {
+		/**
+		 * Description for the attachment, as it exists in the database. Only present when using the 'edit' context.
+		 */
+		raw?: string;
+		/**
+		 * HTML description for the attachment, transformed for display.
+		 */
+		rendered: string;
+	};
+	/**
+	 * Attachment type.
+	 */
+	media_type: 'image' | 'file';
```

Comment:
Can't we also have `video`?

Reply:
No.

https://github.com/WordPress/wordpress-develop/blob/0ac256d5fb63f8f1187d37a067168132390a10ab/src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php#L845-L847